dask/utils.py

```diff
@@ -843,6 +843,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     if protocol == 'file':
         # Special case parsing file protocol URL on Windows according to:
         # https://msdn.microsoft.com/en-us/library/jj710207.aspx
+        # urlsplit does not work with ?
+        path = urlpath if "?" in urlpath else path
```

Comment:
`?` has a completely different meaning in some urls (e.g. http), but it is not used in a `file` scheme (at least, according to https://tools.ietf.org/html/rfc8089).

So, there are 2 options:
- a valid URI was given (so there was no `?` in it)
- a valid glob was given (so it's not a URI at all).

Thus, there is no urlsplit logic for a path with `?`